motionplan/planManager.go

```diff
@@ -612,6 +634,60 @@ func (pm *planManager) goodPlan(pr *rrtPlanReturn, opt *plannerOptions) (bool, f
 	return false, solutionCost
 }
 
+func (pm *planManager) planToRRTGoalMap(plan Plan, goal spatialmath.Pose) (*rrtMaps, error) {
+	planNodes := make([]node, 0, len(plan))
+	// Build a list of nodes from the plan
+	for _, planStep := range plan {
```

Comment:
Do we not have a helper for this yet?  I feel like this has been coming up a lot lately

Reply:
This is a bit different from the way this has been done in the past; it's one of a set of similar-looking but distinct information flows.